### ISSUE TITLE
Repeated dead lettering results in a crashed queue

### ISSUE THREAD

**Issue by cressie176** [rabbitmq/rabbitmq-server#161]
I'm trying to implement a back-off / retry using the process described [here](https://www.lucidchart.com/invitations/accept/ac47f3bf-c290-47b9-8cb9-cd73d31d043d)

![rabbitmq retry back off and failure](https://cloud.githubusercontent.com/assets/229672/7668701/78071946-fc3e-11e4-8f34-67fc2db1dada.png)

However the delay_queue_10_seconds consistently crashes on the second iteration with the following output. I'm using the [rabbitmq:3-management](https://registry.hub.docker.com/_/rabbitmq/) docker image for my testing. Any idea what's going on and how to workaround?

```
rabbitmq_1 | 
rabbitmq_1 |               RabbitMQ 3.5.1. Copyright (C) 2007-2014 GoPivotal, Inc.
rabbitmq_1 |   ##  ##      Licensed under the MPL.  See http://www.rabbitmq.com/
rabbitmq_1 |   ##  ##
rabbitmq_1 |   ##########  Logs: tty
rabbitmq_1 |   ######  ##        /var/log/rabbitmq/my-rabbit-sasl.log
rabbitmq_1 |   ##########
rabbitmq_1 |               Starting broker...
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:08 ===
rabbitmq_1 | Starting RabbitMQ 3.5.1 on Erlang 17.5
rabbitmq_1 | Copyright (C) 2007-2014 GoPivotal, Inc.
rabbitmq_1 | Licensed under the MPL.  See http://www.rabbitmq.com/
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:08 ===
rabbitmq_1 | node           : my-rabbit@bde4b964e57b
rabbitmq_1 | home dir       : /var/lib/rabbitmq
rabbitmq_1 | config file(s) : /etc/rabbitmq/rabbitmq.config
rabbitmq_1 | cookie hash    : niW6EfQVimO2vPi+NEOyZQ==
rabbitmq_1 | log            : tty
rabbitmq_1 | sasl log       : /var/log/rabbitmq/my-rabbit-sasl.log
rabbitmq_1 | database dir   : /var/lib/rabbitmq/mnesia/my-rabbit
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:09 ===
rabbitmq_1 | Limiting to approx 1048476 file handles (943626 sockets)
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:09 ===
rabbitmq_1 |     application: mnesia
rabbitmq_1 |     exited: stopped
rabbitmq_1 |     type: temporary
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Priority queues enabled, real BQ is rabbit_variable_queue
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Memory limit set to 803MB of 2007MB total.
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Disk free limit set to 50MB
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | msg_store_transient: using rabbit_msg_store_ets_index to provide index
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | msg_store_persistent: using rabbit_msg_store_ets_index to provide index
rabbitmq_1 | 
rabbitmq_1 | =WARNING REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | msg_store_persistent: rebuilding indices from scratch
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Adding vhost '/'
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Creating user 'guest'
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Setting user tags for user 'guest' to [administrator]
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Setting permissions for 'guest' in '/' to '.*', '.*', '.*'
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | started TCP Listener on [::]:5672
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Management plugin started. Port: 15672
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Statistics database started.
rabbitmq_1 |  completed with 6 plugins.
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:10 ===
rabbitmq_1 | Server startup complete; 6 plugins started.
rabbitmq_1 |  * rabbitmq_management
rabbitmq_1 |  * rabbitmq_web_dispatch
rabbitmq_1 |  * webmachine
rabbitmq_1 |  * mochiweb
rabbitmq_1 |  * rabbitmq_management_agent
rabbitmq_1 |  * amqp_client
rabbitmq_1 | 
rabbitmq_1 | =INFO REPORT==== 17-May-2015::01:06:18 ===
rabbitmq_1 | accepting AMQP connection <0.368.0> (10.0.2.2:57052 -> 172.17.0.41:5672)
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | ** Generic server <0.388.0> terminating
rabbitmq_1 | ** Last message in was {drop_expired,1}
rabbitmq_1 | ** When Server state == {q,
rabbitmq_1 |                          {amqqueue,
rabbitmq_1 |                           {resource,<<"/">>,queue,
rabbitmq_1 |                            <<"delay_queue_10_seconds">>},
rabbitmq_1 |                           true,false,none,
rabbitmq_1 |                           [{<<"x-message-ttl">>,short,10000},
rabbitmq_1 |                            {<<"x-dead-letter-exchange">>,longstr,
rabbitmq_1 |                             <<"retry_exchange">>}],
rabbitmq_1 |                           <0.388.0>,[],[],[],undefined,[],[],live},
rabbitmq_1 |                          none,false,rabbit_priority_queue,
rabbitmq_1 |                          {passthrough,rabbit_variable_queue,
rabbitmq_1 |                           {vqstate,
rabbitmq_1 |                            {0,{[],[]}},
rabbitmq_1 |                            {0,{[],[]}},
rabbitmq_1 |                            {delta,undefined,0,undefined},
rabbitmq_1 |                            {0,{[],[]}},
rabbitmq_1 |                            {1,
rabbitmq_1 |                             {[],
rabbitmq_1 |                              [{msg_status,1,
rabbitmq_1 |                                <<198,122,81,14,106,190,168,194,59,229,236,71,
rabbitmq_1 |                                  169,235,22,94>>,
rabbitmq_1 |                                {basic_message,
rabbitmq_1 |                                 {resource,<<"/">>,exchange,
rabbitmq_1 |                                  <<"delay_exchange">>},
rabbitmq_1 |                                 [<<"retry.original_queue">>],
rabbitmq_1 |                                 {content,60,
rabbitmq_1 |                                  {'P_basic',undefined,undefined,
rabbitmq_1 |                                   [{<<"delay">>,short,10000},
rabbitmq_1 |                                    {<<"rascal">>,table,
rabbitmq_1 |                                     [{<<"originalQueue">>,longstr,
rabbitmq_1 |                                       <<"original_queue">>},
rabbitmq_1 |                                      {<<"forwarded">>,byte,2},
rabbitmq_1 |                                      {<<"originalExchange">>,longstr,
rabbitmq_1 |                                       <<"original_exchange">>},
rabbitmq_1 |                                      {<<"originalRoutingKey">>,longstr,
rabbitmq_1 |                                       <<"retry.original_queue">>}]},
rabbitmq_1 |                                    {<<"x-death">>,array,
rabbitmq_1 |                                     [{table,
rabbitmq_1 |                                       [{<<"count">>,byte,1},
rabbitmq_1 |                                        {<<"reason">>,longstr,<<"expired">>},
rabbitmq_1 |                                        {<<"queue">>,longstr,
rabbitmq_1 |                                         <<"delay_queue_10_seconds">>},
rabbitmq_1 |                                        {<<"time">>,timestamp,1431824813},
rabbitmq_1 |                                        {<<"exchange">>,longstr,
rabbitmq_1 |                                         <<"delay_exchange">>},
rabbitmq_1 |                                        {<<"routing-keys">>,array,
rabbitmq_1 |                                         [{longstr,
rabbitmq_1 |                                           <<"retry.original_queue">>}]}]}]}],
rabbitmq_1 |                                   2,undefined,undefined,undefined,undefined,
rabbitmq_1 |                                   undefined,undefined,undefined,undefined,
rabbitmq_1 |                                   undefined,undefined},
rabbitmq_1 |                                  <<48,0,0,0,1,56,5,100,101,108,97,121,115,39,
rabbitmq_1 |                                    16,6,114,97,115,99,97,108,70,0,0,0,128,13,
rabbitmq_1 |                                    111,114,105,103,105,110,97,108,81,117,101,
rabbitmq_1 |                                    117,101,83,0,0,0,14,111,114,105,103,105,110,
rabbitmq_1 |                                    97,108,95,113,117,101,117,101,9,102,111,114,
rabbitmq_1 |                                    119,97,114,100,101,100,98,2,16,111,114,105,
rabbitmq_1 |                                    103,105,110,97,108,69,120,99,104,97,110,103,
rabbitmq_1 |                                    101,83,0,0,0,17,111,114,105,103,105,110,97,
rabbitmq_1 |                                    108,95,101,120,99,104,97,110,103,101,18,111,
rabbitmq_1 |                                    114,105,103,105,110,97,108,82,111,117,116,
rabbitmq_1 |                                    105,110,103,75,101,121,83,0,0,0,20,114,101,
rabbitmq_1 |                                    116,114,121,46,111,114,105,103,105,110,97,
rabbitmq_1 |                                    108,95,113,117,101,117,101,7,120,45,100,101,
rabbitmq_1 |                                    97,116,104,65,0,0,0,150,70,0,0,0,145,5,99,
rabbitmq_1 |                                    111,117,110,116,98,1,6,114,101,97,115,111,
rabbitmq_1 |                                    110,83,0,0,0,7,101,120,112,105,114,101,100,
rabbitmq_1 |                                    5,113,117,101,117,101,83,0,0,0,22,100,101,
rabbitmq_1 |                                    108,97,121,95,113,117,101,117,101,95,49,48,
rabbitmq_1 |                                    95,115,101,99,111,110,100,115,4,116,105,109,
rabbitmq_1 |                                    101,84,0,0,0,0,85,87,233,173,8,101,120,99,
rabbitmq_1 |                                    104,97,110,103,101,83,0,0,0,14,100,101,108,
rabbitmq_1 |                                    97,121,95,101,120,99,104,97,110,103,101,12,
rabbitmq_1 |                                    114,111,117,116,105,110,103,45,107,101,121,
rabbitmq_1 |                                    115,65,0,0,0,25,83,0,0,0,20,114,101,116,114,
rabbitmq_1 |                                    121,46,111,114,105,103,105,110,97,108,95,
rabbitmq_1 |                                    113,117,101,117,101,2>>,
rabbitmq_1 |                                  rabbit_framing_amqp_0_9_1,
rabbitmq_1 |                                  [<<"Test">>]},
rabbitmq_1 |                                 <<198,122,81,14,106,190,168,194,59,229,236,71,
rabbitmq_1 |                                   169,235,22,94>>,
rabbitmq_1 |                                 true},
rabbitmq_1 |                                true,false,false,true,queue_index,
rabbitmq_1 |                                {message_properties,1431824823202163,false,
rabbitmq_1 |                                 4}}]}},
rabbitmq_1 |                            2,
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            {qistate,
rabbitmq_1 |                             "/var/lib/rabbitmq/mnesia/my-rabbit/queues/3XSMVIFXMKD5LZ8A8OAHSI5C3",
rabbitmq_1 |                             {{dict,0,16,16,8,80,48,
rabbitmq_1 |                               {[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                []},
rabbitmq_1 |                               {{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                 []}}},
rabbitmq_1 |                              [{segment,0,
rabbitmq_1 |                                "/var/lib/rabbitmq/mnesia/my-rabbit/queues/3XSMVIFXMKD5LZ8A8OAHSI5C3/0.idx",
rabbitmq_1 |                                {array,16384,0,undefined,100000},
rabbitmq_1 |                                1}]},
rabbitmq_1 |                             #Ref<0.0.0.2626>,0,65536,
rabbitmq_1 |                             #Fun<rabbit_variable_queue.2.117761292>,
rabbitmq_1 |                             #Fun<rabbit_variable_queue.3.48316793>,
rabbitmq_1 |                             {0,nil},
rabbitmq_1 |                             {0,nil}},
rabbitmq_1 |                            {{client_msstate,msg_store_persistent,
rabbitmq_1 |                              <<199,191,139,31,172,75,95,166,68,36,46,251,147,
rabbitmq_1 |                                213,204,106>>,
rabbitmq_1 |                              {dict,0,16,16,8,80,48,
rabbitmq_1 |                               {[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                []},
rabbitmq_1 |                               {{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                 []}}},
rabbitmq_1 |                              {state,307272,
rabbitmq_1 |                               "/var/lib/rabbitmq/mnesia/my-rabbit/msg_store_persistent"},
rabbitmq_1 |                              rabbit_msg_store_ets_index,
rabbitmq_1 |                              "/var/lib/rabbitmq/mnesia/my-rabbit/msg_store_persistent",
rabbitmq_1 |                              <0.286.0>,311369,303175,315466,319563},
rabbitmq_1 |                             {client_msstate,msg_store_transient,
rabbitmq_1 |                              <<75,133,213,101,163,234,164,56,116,27,70,169,231,
rabbitmq_1 |                                206,138,195>>,
rabbitmq_1 |                              {dict,0,16,16,8,80,48,
rabbitmq_1 |                               {[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                []},
rabbitmq_1 |                               {{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                 []}}},
rabbitmq_1 |                              {state,286787,
rabbitmq_1 |                               "/var/lib/rabbitmq/mnesia/my-rabbit/msg_store_transient"},
rabbitmq_1 |                              rabbit_msg_store_ets_index,
rabbitmq_1 |                              "/var/lib/rabbitmq/mnesia/my-rabbit/msg_store_transient",
rabbitmq_1 |                              <0.281.0>,290884,282690,294981,299078}},
rabbitmq_1 |                            true,0,1,4,0,1,4,infinity,1,0,0,4,0,0,
rabbitmq_1 |                            {rates,0.07914896001660765,0.07094865606545264,
rabbitmq_1 |                             0.07094865606545264,0.07094865606545264,
rabbitmq_1 |                             {1431,824816,439037}},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            0,0,0,2}},
rabbitmq_1 |                          {state,
rabbitmq_1 |                           {queue,[],[],0},
rabbitmq_1 |                           {inactive,1431824803190030,24412513,1.0}},
rabbitmq_1 |                          undefined,undefined,undefined,undefined,
rabbitmq_1 |                          {state,fine,5000,undefined},
rabbitmq_1 |                          {0,nil},
rabbitmq_1 |                          10000,
rabbitmq_1 |                          {erlang,#Ref<0.0.0.2814>},
rabbitmq_1 |                          1431824823202163,
rabbitmq_1 |                          {state,
rabbitmq_1 |                           {dict,0,16,16,8,80,48,
rabbitmq_1 |                            {[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]},
rabbitmq_1 |                            {{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                              []}}},
rabbitmq_1 |                           delegate},
rabbitmq_1 |                          {resource,<<"/">>,exchange,<<"retry_exchange">>},
rabbitmq_1 |                          undefined,undefined,undefined,1,running}
rabbitmq_1 | ** Reason for termination == 
rabbitmq_1 | ** {{case_clause,{value,{<<"count">>,byte,1}}},
rabbitmq_1 |     [{rabbit_dead_letter,x_death_event_key,3,[]},
rabbitmq_1 |      {rabbit_dead_letter,update_x_death_header,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,'-make_msg/5-fun-2-',8,[]},
rabbitmq_1 |      {rabbit_basic,map_headers,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,make_msg,5,[]},
rabbitmq_1 |      {rabbit_dead_letter,publish,5,[]},
rabbitmq_1 |      {rabbit_amqqueue_process,'-dead_letter_msgs/4-fun-0-',7,[]},
rabbitmq_1 |      {rabbit_variable_queue,fetchwhile,4,[]}]}
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | Restarting crashed queue 'delay_queue_10_seconds' in vhost '/'.
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | ** Generic server <0.470.0> terminating
rabbitmq_1 | ** Last message in was {'$gen_cast',init}
rabbitmq_1 | ** When Server state == {q,{amqqueue,
rabbitmq_1 |                                {resource,<<"/">>,queue,
rabbitmq_1 |                                    <<"delay_queue_10_seconds">>},
rabbitmq_1 |                                true,false,none,
rabbitmq_1 |                                [{<<"x-message-ttl">>,short,10000},
rabbitmq_1 |                                 {<<"x-dead-letter-exchange">>,longstr,
rabbitmq_1 |                                  <<"retry_exchange">>}],
rabbitmq_1 |                                <0.470.0>,[],[],[],undefined,[],[],crashed},
rabbitmq_1 |                            none,false,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {queue,[],[],0},
rabbitmq_1 |                                {active,1431824823209199,1.0}},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,
rabbitmq_1 |                            {state,fine,5000,undefined},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            undefined,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {dict,0,16,16,8,80,48,
rabbitmq_1 |                                    {[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                     [],[]},
rabbitmq_1 |                                    {{[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                      [],[],[]}}},
rabbitmq_1 |                                delegate},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,0,running}
rabbitmq_1 | ** Reason for termination == 
rabbitmq_1 | ** {{case_clause,{value,{<<"count">>,byte,1}}},
rabbitmq_1 |     [{rabbit_dead_letter,x_death_event_key,3,[]},
rabbitmq_1 |      {rabbit_dead_letter,update_x_death_header,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,'-make_msg/5-fun-2-',8,[]},
rabbitmq_1 |      {rabbit_basic,map_headers,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,make_msg,5,[]},
rabbitmq_1 |      {rabbit_dead_letter,publish,5,[]},
rabbitmq_1 |      {rabbit_amqqueue_process,'-dead_letter_msgs/4-fun-0-',7,[]},
rabbitmq_1 |      {rabbit_variable_queue,fetchwhile,4,[]}]}
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | Restarting crashed queue 'delay_queue_10_seconds' in vhost '/'.
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | ** Generic server <0.472.0> terminating
rabbitmq_1 | ** Last message in was {'$gen_cast',init}
rabbitmq_1 | ** When Server state == {q,{amqqueue,
rabbitmq_1 |                                {resource,<<"/">>,queue,
rabbitmq_1 |                                    <<"delay_queue_10_seconds">>},
rabbitmq_1 |                                true,false,none,
rabbitmq_1 |                                [{<<"x-message-ttl">>,short,10000},
rabbitmq_1 |                                 {<<"x-dead-letter-exchange">>,longstr,
rabbitmq_1 |                                  <<"retry_exchange">>}],
rabbitmq_1 |                                <0.472.0>,[],[],[],undefined,[],[],live},
rabbitmq_1 |                            none,false,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {queue,[],[],0},
rabbitmq_1 |                                {active,1431824823219033,1.0}},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,
rabbitmq_1 |                            {state,fine,5000,undefined},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            undefined,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {dict,0,16,16,8,80,48,
rabbitmq_1 |                                    {[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                     [],[]},
rabbitmq_1 |                                    {{[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                      [],[],[]}}},
rabbitmq_1 |                                delegate},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,0,running}
rabbitmq_1 | ** Reason for termination == 
rabbitmq_1 | ** {{case_clause,{value,{<<"count">>,byte,1}}},
rabbitmq_1 |     [{rabbit_dead_letter,x_death_event_key,3,[]},
rabbitmq_1 |      {rabbit_dead_letter,update_x_death_header,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,'-make_msg/5-fun-2-',8,[]},
rabbitmq_1 |      {rabbit_basic,map_headers,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,make_msg,5,[]},
rabbitmq_1 |      {rabbit_dead_letter,publish,5,[]},
rabbitmq_1 |      {rabbit_amqqueue_process,'-dead_letter_msgs/4-fun-0-',7,[]},
rabbitmq_1 |      {rabbit_variable_queue,fetchwhile,4,[]}]}
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | Restarting crashed queue 'delay_queue_10_seconds' in vhost '/'.
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | ** Generic server <0.474.0> terminating
rabbitmq_1 | ** Last message in was {'$gen_cast',init}
rabbitmq_1 | ** When Server state == {q,{amqqueue,
rabbitmq_1 |                                {resource,<<"/">>,queue,
rabbitmq_1 |                                    <<"delay_queue_10_seconds">>},
rabbitmq_1 |                                true,false,none,
rabbitmq_1 |                                [{<<"x-message-ttl">>,short,10000},
rabbitmq_1 |                                 {<<"x-dead-letter-exchange">>,longstr,
rabbitmq_1 |                                  <<"retry_exchange">>}],
rabbitmq_1 |                                <0.474.0>,[],[],[],undefined,[],[],live},
rabbitmq_1 |                            none,false,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {queue,[],[],0},
rabbitmq_1 |                                {active,1431824823228335,1.0}},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,
rabbitmq_1 |                            {state,fine,5000,undefined},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            undefined,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {dict,0,16,16,8,80,48,
rabbitmq_1 |                                    {[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                     [],[]},
rabbitmq_1 |                                    {{[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                      [],[],[]}}},
rabbitmq_1 |                                delegate},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,0,running}
rabbitmq_1 | ** Reason for termination == 
rabbitmq_1 | ** {{case_clause,{value,{<<"count">>,byte,1}}},
rabbitmq_1 |     [{rabbit_dead_letter,x_death_event_key,3,[]},
rabbitmq_1 |      {rabbit_dead_letter,update_x_death_header,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,'-make_msg/5-fun-2-',8,[]},
rabbitmq_1 |      {rabbit_basic,map_headers,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,make_msg,5,[]},
rabbitmq_1 |      {rabbit_dead_letter,publish,5,[]},
rabbitmq_1 |      {rabbit_amqqueue_process,'-dead_letter_msgs/4-fun-0-',7,[]},
rabbitmq_1 |      {rabbit_variable_queue,fetchwhile,4,[]}]}
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | Restarting crashed queue 'delay_queue_10_seconds' in vhost '/'.
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | ** Generic server <0.476.0> terminating
rabbitmq_1 | ** Last message in was {'$gen_cast',init}
rabbitmq_1 | ** When Server state == {q,{amqqueue,
rabbitmq_1 |                                {resource,<<"/">>,queue,
rabbitmq_1 |                                    <<"delay_queue_10_seconds">>},
rabbitmq_1 |                                true,false,none,
rabbitmq_1 |                                [{<<"x-message-ttl">>,short,10000},
rabbitmq_1 |                                 {<<"x-dead-letter-exchange">>,longstr,
rabbitmq_1 |                                  <<"retry_exchange">>}],
rabbitmq_1 |                                <0.476.0>,[],[],[],undefined,[],[],live},
rabbitmq_1 |                            none,false,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {queue,[],[],0},
rabbitmq_1 |                                {active,1431824823234835,1.0}},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,
rabbitmq_1 |                            {state,fine,5000,undefined},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            undefined,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {dict,0,16,16,8,80,48,
rabbitmq_1 |                                    {[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                     [],[]},
rabbitmq_1 |                                    {{[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                      [],[],[]}}},
rabbitmq_1 |                                delegate},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,0,running}
rabbitmq_1 | ** Reason for termination == 
rabbitmq_1 | ** {{case_clause,{value,{<<"count">>,byte,1}}},
rabbitmq_1 |     [{rabbit_dead_letter,x_death_event_key,3,[]},
rabbitmq_1 |      {rabbit_dead_letter,update_x_death_header,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,'-make_msg/5-fun-2-',8,[]},
rabbitmq_1 |      {rabbit_basic,map_headers,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,make_msg,5,[]},
rabbitmq_1 |      {rabbit_dead_letter,publish,5,[]},
rabbitmq_1 |      {rabbit_amqqueue_process,'-dead_letter_msgs/4-fun-0-',7,[]},
rabbitmq_1 |      {rabbit_variable_queue,fetchwhile,4,[]}]}
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | Restarting crashed queue 'delay_queue_10_seconds' in vhost '/'.
rabbitmq_1 | 
rabbitmq_1 | =ERROR REPORT==== 17-May-2015::01:07:03 ===
rabbitmq_1 | ** Generic server <0.478.0> terminating
rabbitmq_1 | ** Last message in was {'$gen_cast',init}
rabbitmq_1 | ** When Server state == {q,{amqqueue,
rabbitmq_1 |                                {resource,<<"/">>,queue,
rabbitmq_1 |                                    <<"delay_queue_10_seconds">>},
rabbitmq_1 |                                true,false,none,
rabbitmq_1 |                                [{<<"x-message-ttl">>,short,10000},
rabbitmq_1 |                                 {<<"x-dead-letter-exchange">>,longstr,
rabbitmq_1 |                                  <<"retry_exchange">>}],
rabbitmq_1 |                                <0.478.0>,[],[],[],undefined,[],[],live},
rabbitmq_1 |                            none,false,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {queue,[],[],0},
rabbitmq_1 |                                {active,1431824823241172,1.0}},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,
rabbitmq_1 |                            {state,fine,5000,undefined},
rabbitmq_1 |                            {0,nil},
rabbitmq_1 |                            undefined,undefined,undefined,
rabbitmq_1 |                            {state,
rabbitmq_1 |                                {dict,0,16,16,8,80,48,
rabbitmq_1 |                                    {[],[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                     [],[]},
rabbitmq_1 |                                    {{[],[],[],[],[],[],[],[],[],[],[],[],[],
rabbitmq_1 |                                      [],[],[]}}},
rabbitmq_1 |                                delegate},
rabbitmq_1 |                            undefined,undefined,undefined,undefined,0,running}
rabbitmq_1 | ** Reason for termination == 
rabbitmq_1 | ** {{case_clause,{value,{<<"count">>,byte,1}}},
rabbitmq_1 |     [{rabbit_dead_letter,x_death_event_key,3,[]},
rabbitmq_1 |      {rabbit_dead_letter,update_x_death_header,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,'-make_msg/5-fun-2-',8,[]},
rabbitmq_1 |      {rabbit_basic,map_headers,2,[]},
rabbitmq_1 |      {rabbit_dead_letter,make_msg,5,[]},
rabbitmq_1 |      {rabbit_dead_letter,publish,5,[]},
rabbitmq_1 |      {rabbit_amqqueue_process,'-dead_letter_msgs/4-fun-0-',7,[]},
rabbitmq_1 |      {rabbit_variable_queue,fetchwhile,4,[]}]}
```

**Comment by cressie176** [comment]
Deleting the x-death header between prior to republishing resolves the issue. I'd rather not do this as it's valuable information. It's easy for me to produce a test case if you don't mind running a node application.

**Comment by michaelklishin** [comment]
See #152, released in `3.5.2`.
